src/porepy/grids/grid.py

```diff
@@ -880,41 +881,123 @@ def update_boundary_node_tag(self) -> None:
+        func: Optional[Callable] = None,
+    ) -> np.ndarray:
         """Computes the cell diameters.
 
         Parameters:
             cn: ``default=None``
-                Cell-to-nodes map, already computed previously.
-                If None, a call to :meth:`cell_nodes` is provided.
+                Cell-to-nodes map, already computed previously. If None, a call to
+                :meth:`cell_nodes` is provided.
+
+            cell_wise:
+                If True, returns the cell diameters. If False, the parameter ``func``
+                must be provided to specify the aggregation function.
+
+                The requirement that ``func`` is provided is linked to technical
+                limitations of the implementation.
+
+            func:
+                A function to apply to the computed diameters. Must be provided if
+                ``cell_wise`` is False.
+
+        Raises:
+            ValueError: If ``func`` is not provided when ``cell_wise`` is False.
 
         Returns:
-            Values of the cell diameter for each cell, ``(shape=(num_cells))``.
+
+            If ``cell_wise`` is True, the cell diameter for each cell,
+            ``(shape=(num_cells))``. If ``cell_wise`` is False, the maximum diameter of
+            all cells.
 
             If the dimension of the grid is zero, returns 0.
 
         """
         if self.dim == 0:
             return np.zeros(1)
 
+        if cell_wise is False and func is None:
+            raise ValueError("func must be provided if cell_wise is False")
+
+        if cn is None:
+            cn = self.cell_nodes()
+
         def comb(n):
+            # Helper function to get all combinations of two elements in n.
             return np.fromiter(
                 itertools.chain.from_iterable(itertools.combinations(n, 2)), n.dtype
             ).reshape((2, -1), order="F")
 
         def diam(n):
+            # Helper function to compute the diameter of a set of nodes.
             return np.amax(
                 np.linalg.norm(self.nodes[:, n[0, :]] - self.nodes[:, n[1, :]], axis=0)
             )
 
-        if cn is None:
-            cn = self.cell_nodes()
-        return np.array(
-            [
-                diam(comb(cn.indices[cn.indptr[c] : cn.indptr[c + 1]]))
-                for c in np.arange(self.num_cells)
-            ]
-        )
+        if cell_wise:
+            # Compute the diameter for each cell by 1) getting all node combinations for
+            # the cell and 2) computing the diameter of the cell. This implementation
+            # can likely be optimized using techniques similar to the ones used in
+            # the below else, but implementing this has not been a priority.
+            cell_diameters = np.array(
+                [
+                    diam(comb(cn.indices[cn.indptr[c] : cn.indptr[c + 1]]))
+                    for c in np.arange(self.num_cells)
+                ]
+            )
+            return cell_diameters
+
+        else:
```

Comment:
This was not super-easy code to understand. However, having invested a bit of time, my impression is 1. this seems plausible and 2. neat solution. Thanks for the documentation.